src/app/shared/abstracts/base-product/base-product.component.ts

```diff
@@ -0,0 +1,67 @@
+import { Component, Input } from "@angular/core";
+import { FormBuilder, Validators } from "@angular/forms";
+
+import { Product } from "@shared/models/classes/product.class";
+import { Roast } from "@shared/models/types/roast.type";
+import { Weight } from "@shared/models/types/weight.type";
+import { CustomOption } from "@shared/models/types/custom-option.interface";
+
+import { ROAST_OPTIONS, WEIGHT_OPTIONS } from "@utils/constants/form-options";
+
+@Component({
+    selector: "app-base-product",
+    templateUrl: "",
+    styleUrls: [""]
+})
```

Comment:
here, we can remove most of the keys in the @Component decorator. So we can leave as ({ template: "" })

otherwise, the component errors, the module errors and the compilation fails